akka-cluster/src/multi-jvm/scala/akka/cluster/MultiTeamClusterSpec.scala

```diff
@@ -0,0 +1,140 @@
+
+abstract class MultiTeamSpec
+  extends MultiNodeSpec(MultiTeamMultiJvmSpec)
+  with MultiNodeClusterSpec {
+
+  import MultiTeamMultiJvmSpec._
+
+  "A cluster with multiple cluster teams" must {
+    "be able to form" in {
+
+      runOn(first) {
+        cluster.join(first)
+      }
+      runOn(second, third, fourth) {
+        cluster.join(first)
+      }
+      enterBarrier("form-cluster-join-attempt")
+
+      runOn(first, second, third, fourth) {
+        within(15.seconds) {
+          awaitAssert(clusterView.members.count(_.status == MemberStatus.Up) should ===(4))
+        }
+      }
+
+      enterBarrier("cluster started")
+    }
+
+    "have a leader per team" in {
+      runOn(first, second) {
+        cluster.settings.Team should ===("dc1")
+        clusterView.leader shouldBe defined
+        val dc1 = Set(address(first), address(second))
+        dc1.contains(clusterView.leader.get) should ===(true)
+      }
+      runOn(third, fourth) {
+        cluster.settings.Team should ===("dc2")
+        clusterView.leader shouldBe defined
+        val dc2 = Set(address(third), address(fourth))
+        dc2.contains(clusterView.leader.get) should ===(true)
+      }
+
+      enterBarrier("leader per team")
+    }
+
+    "be able to have team member changes while there is inter-team unreachability" in within(10.seconds) {
+      runOn(first) {
+        testConductor.blackhole(first, third, Direction.Both)
+      }
+      runOn(first, second, third, fourth) {
+        within(remaining) {
+          awaitAssert(clusterView.unreachableMembers.nonEmpty)
+        }
+      }
+      enterBarrier("inter-team unreachability")
+
+      runOn(fifth) {
+        cluster.join(first)
+      }
+
+      // should be able to join and become up since the
+      // unreachable is between dc1 and dc2
+      within(15.seconds) {
+        awaitAssert(clusterView.members.count(_.status == MemberStatus.Up) should ===(5))
+      }
+
+      runOn(first) {
+        testConductor.passThrough(first, third, Direction.Both)
+      }
+      runOn(first, second, third, fourth) {
+        within(remaining) {
+          awaitAssert(clusterView.unreachableMembers.isEmpty)
+        }
+      }
+      enterBarrier("end")
+    }
+
+    "be able to have team member changes while there is unreachability in another team" in within(10.seconds) {
+      runOn(first) {
+        testConductor.blackhole(first, second, Direction.Both)
+      }
+      runOn(first, second, third, fourth) {
+        within(remaining) {
+          awaitAssert(clusterView.unreachableMembers.nonEmpty)
+        }
+      }
+      enterBarrier("other-team-internal-unreachable")
+
+      runOn(fifth) {
+        cluster.leave(first)
+      }
+
+      // should be able to leave and become removed
+      // since the unreachable nodes are inside of dc1
+      within(15.seconds) {
+        awaitAssert(clusterView.members.find(_.status == MemberStatus.Removed))
+      }
```

Comment:
perhaps add to this test that you leave in `dc1` also and verify that it is not removed (will require one more node)